### ISSUE TITLE
Ansi-tests: Adjust array should work on non adjustable arrays

### ISSUE THREAD

**Issue by kpoeck** [drmeister/clasp#424]
Looking at http://www.lispworks.com/documentation/HyperSpec/Body/f_adjust.htm it doesn't seem that the array argument to adjust-array must be actually adjustable.

So the following is supposed to work:
````
2 > (adjust-array (make-array 5 :adjustable nil :initial-element 42) 7 :initial-element 8)
#(42 42 42 42 42 8 8)
````

but on current clasp:
````
COMMON-LISP-USER> (adjust-array (make-array 5 :adjustable nil :initial-element 42) 7 :initial-element 8)

Condition of type: SIMPLE-PROGRAM-ERROR
In REPLACE-ARRAY - array is not adjustable

Available restarts:
(use :r1 to invoke restart 1)

1. (RESTART-TOPLEVEL) Go back to Top-Level REPL.

Broken at frame[0] NIL. In: #<PROCESS TOP-LEVEL @0x10aaa7e09>.
COMMON-LISP-USER>> 
````

**Comment by kpoeck** [comment]
I think the fix is a simple as:
````
(defun adjust-array (array new-dimensions
                     &rest r
		     &key (element-type (array-element-type array))
			  initial-element
			  initial-contents
			  fill-pointer
			  displaced-to
			  displaced-index-offset)
  "Args: (array dimensions
       &key (element-type (array-element-type array))
            initial-element (initial-contents nil) (fill-pointer nil)
            (displaced-to nil) (displaced-index-offset 0))
Adjusts the dimensions of ARRAY to the given DIMENSIONS.  ARRAY -- kpoeck -- may be an
adjustable array."
  (declare (ignore initial-element
                   displaced-index-offset))
  (when (integerp new-dimensions)
        (setq new-dimensions (list new-dimensions)))
  ;; FILL-POINTER = NIL means use the old value of the fill pointer
  ;; Cannot set a fill pointer for an array that does not have any.
  (if (array-has-fill-pointer-p array)
      (unless fill-pointer
	(setf r (list* :fill-pointer (fill-pointer array) r)))
      (when fill-pointer
	(error 'simple-type-error
	       :datum array
	       :expected-type '(satisfies array-has-fill-pointer-p)
	       :format-control "You supplied a fill pointer for an array without it.")))
  (let ((x (apply #'make-array new-dimensions :adjustable t :element-type element-type r)))
    (declare (array x))
    (unless (or displaced-to initial-contents)
      (copy-array-contents x array))
    ;Kpoeck
    (if (adjustable-array-p array)
        (sys:replace-array array x)
        x)
    ))
````

Will do a PR

**Comment by kpoeck** [comment]
Fixed by PR #425

**Comment by kpoeck** [comment]
Fixed in dev with the merged PR
